Neos.Media.Browser/Classes/Controller/UsageController.php

```diff
@@ -196,24 +220,55 @@ public function relatedNodesAction(AssetInterface $asset)
         ]);
     }
 
-    private function getLabelForInaccessibleWorkspace(Workspace $workspace): string
+    /**
+     * @return array{title: WorkspaceTitle|null, owner: User|null, personalWorkspace: bool, privateWorkspace: bool}
+     */
+    private function getRelevantMetadataFromInaccessibleWorkspace(?Workspace $workspace): array
     {
-        /*
-        TODO: Needs to get re-implemented for Neos 9.
-              See: https://github.com/neos/neos-development-collection/pull/5182
+        $structuredReturn = [
+            'title' => null,
+            'owner' => null,
+            'personalWorkspace' => false,
+            'privateWorkspace' => false,
+        ];
+
+        if ($workspace === null) {
+            return $structuredReturn;
+        }
 
         $currentAccount = $this->securityContext->getAccount();
 
         if ($currentAccount != null && $this->privilegeManager->isPrivilegeTargetGranted('Neos.Media.Browser:WorkspaceName')) {
-            if ($workspace->isPrivateWorkspace()) {
-                $owner = $workspace->getOwner();
-                return '(' . $owner->getLabel() . ')';
+            $contentRepositoryId = SiteDetectionResult::fromRequest($this->request->getHttpRequest())->contentRepositoryId;
+            $contentRepository = $this->contentRepositoryRegistry->get($contentRepositoryId);
+
+            $workspaceMetadata = $this->workspaceService->getWorkspaceMetadata($contentRepository->id, $workspace->workspaceName);
+            $workspaceOwner = $workspaceMetadata->ownerUserId
+                ? $this->domainUserService->findUserById($workspaceMetadata->ownerUserId)
+                : null;
+
+            $roleAssignments = $this->workspaceService->getWorkspaceRoleAssignments(
+                $contentRepositoryId,
+                $workspace->workspaceName
+            );
+            $relatedUser = null;
+            foreach ($roleAssignments as $roleAssignment) {
+                if (($roleAssignment->role->value !== WorkspaceRole::VIEWER->value) && ($roleAssignment->subject->type->value === WorkspaceRoleSubjectType::USER->value)) {
+                    $relatedUser = $this->domainUserService->findUserById(UserId::fromString($roleAssignment->subject->value));
+                    break;
+                }
+            }
+
+            if ($workspaceMetadata->classification->value === WorkspaceClassification::PERSONAL->value) {
+                $structuredReturn['owner'] = $workspaceOwner;
```

Comment:
As far as I can see, we only use the title of the workspace user, do we need to pass the whole object to the view or could we just use the label here?